### PR TITLE
Photo testing

### PR DESCRIPTION
**Pull request by lucamoroz**
Merge the updated testing, which now includes alterations testing, into CommonDemo

### CODE REVIEW

**Comment by lucamoroz** [comment]
This pull request has priority for testing the application, allowing us to run tests on modified photo (e.g. rotated, cropped or blurred) and discover weaknesses of the OCR

File: app/src/main/java/unipd/se18/ocrcamera/AdapterTestElement.java [hunk 1]
```diff
@@ -64,79 +66,101 @@ public long getItemId(int position) {
...
-            StringBuilder assignedTags = new StringBuilder();
-            for(String tag: entries[position].getTags()) {
-                assignedTags.append(tag).append(", ");
+        Button viewAlterationsButton = convertView.findViewById(R.id.view_alterations_button);
+        viewAlterationsButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Intent i = new Intent(context,TestAlterationsActivity.class);
+                TestAlterationsActivity.entry = entries[position];
+                context.startActivity(i);
```
Comment (stefanoromanello) [inline app/src/main/java/unipd/se18/ocrcamera/AdapterTestElement.java:82]: You are passing to TestAlterationsActivity the value entries[position] as TestElement. Couldn't you just implement into TestResultActivity this part of TestAlterationsActivity ??
```
        // Set the Tags text
        TextView tags = convertView.findViewById(R.id.tags_view);
        StringBuilder assignedTags = new StringBuilder();

        for(String tag: entry.getAlterationTags(entry.getAlterationsNames()[position])) {
            assignedTags.append(tag).append(", ");
        }
        tags.setText(assignedTags.toString());
```
In this way we don't use another activity and adapter.

Comment (stefanoromanello) [reply app/src/main/java/unipd/se18/ocrcamera/AdapterTestElement.java:82]: where 
`entry=entries[position];`

Comment (stefanoromanello) [reply app/src/main/java/unipd/se18/ocrcamera/AdapterTestElement.java:82]: Oh ok, you would like to show also the score and the extracted text of the alterations

Comment (PietroPrandini) [reply app/src/main/java/unipd/se18/ocrcamera/AdapterTestElement.java:82]: Thanks for reviewing.
Your comment has showed me a thing that I've missing.
So yes, I'd like to show the details of the alterations in a new specific activity but I've would like also to show a summary of the alterations test. So I've prepared the string for doing that but I've not used for that.
I'll add it soon.
Thanks.

**Review by stefanoromanello — APPROVED** [review]
Ok, for me this works

**Pull request merged**